indico/web/client/js/react/components/pictures/PictureManager.jsx

```diff
@@ -0,0 +1,365 @@
+    if (
+      cameraBackActionRef.current === 'finished' &&
+      (pictureState.picture.uuid === null || pictureState.picture.uuid === value)
+    ) {
+      dispatch({type: 'UPLOAD_FINISHED', picture: pictureState.picture});
+    } else {
+      dispatch({type: 'RESET'});
+    }
+  };
+
+  const Camera = () => (
+    <PictureWebcam
+      cameraRef={cameraRef}
+      onCapture={onCapture}
+      onUserMediaError={onCameraMediaError}
+      backAction={cameraBackAction}
+    />
+  );
+
+  const onOpenCaptureDialog = () => {
+    dispatch({type: 'START_CAPTURING'});
+  };
+
+  const uploadPicture = useCallback(
+    async file => {
+      deleteUploadedPicture();
+      dispatch({type: 'START_UPLOAD', picture: file});
+      const resp = await uploadFile(uploadURL, file, e =>
+        dispatch({
+          type: 'PROGRESS',
+          percent: Math.floor((e.loaded / e.total) * 100),
+        })
+      );
+      if (resp !== null) {
+        dispatch({type: 'UPLOAD_FINISHED', picture: resp});
+        onChange(resp.uuid);
+      } else {
+        dispatch({type: 'FAILED'});
+        setPicturePreview(previewURL);
+        onChange(null);
+      }
+    },
+    [deleteUploadedPicture, onChange, uploadURL, previewURL]
+  );
+
+  const onImageCrop = () => {
+    const imageSrc = cropperRef.current.cropper.getCroppedCanvas().toDataURL();
+    setPicturePreview(imageSrc);
+    fetch(imageSrc)
+      .then(res => res.blob())
+      .then(blob => {
+        const file = new File([blob], 'picture.png', blob);
+        uploadPicture(file);
+      });
+  };
+
+  useEffect(() => {
+    if (isCapturing) {
+      cropperBackActionRef.current = 'camera';
+    } else if (uploadFinished) {
+      cropperBackActionRef.current = 'finished';
+    } else if (isInitialPicture) {
+      cropperBackActionRef.current = 'initial';
+    }
+  }, [isCapturing, uploadFinished, isInitialPicture]);
+
+  const cropBackAction = () => {
+    if (cropperBackActionRef.current === 'camera') {
+      dispatch({type: 'START_CAPTURING'});
+    } else if (
+      cropperBackActionRef.current === 'finished' &&
+      (pictureState.picture.uuid === null || pictureState.picture.uuid === value)
+    ) {
+      dispatch({type: 'UPLOAD_FINISHED', picture: pictureState.picture});
+    } else {
+      dispatch({type: 'RESET'});
+    }
+  };
+
+  const Cropper = () => (
+    <PictureCropper
+      cropperRef={cropperRef}
+      src={pictureState.imageSrc}
+      onCrop={onImageCrop}
+      backAction={cropBackAction}
+    />
+  );
+
+  const onDropAccepted = async ([file]) => {
+    let imageSrc = null;
+    const reader = new FileReader();
+    reader.addEventListener('load', () => {
+      imageSrc = reader.result;
+      dispatch({type: 'START_EDITING', imageSrc});
+    });
+    reader.readAsDataURL(file);
+  };
+
+  const dropzone = useDropzone({
+    onDropAccepted,
+    disabled: disabled || (isUploading || failed || isCapturing || isEditing),
+    accept: ['.png', '.jpg', '.gif'],
```

Comment:
As a suggestion: would it be possible to indicate which file types are accepted on the dropzone UI (i.e. 'accepted files: .jpg, .png, .gif.')?

Reply:
I was passing it explicitly since we wouldn't really want something else that isn't an image to be uploaded here. But in what way would you suggest to indicate this? a comment?

Reply:
In terms of indicating this, I would suggest adding an extra line of text on the dropzone UI.

Reply:
Adding onto this: [this might also be useful](https://ux.stackexchange.com/a/112618).

Reply:
Ah thanks, I see. Good idea, I'll add the extra text